pyanaconda/dbus/template.py

```diff
@@ -99,3 +100,41 @@ def publish(self, object_path):
         """
         DBus.publish_object(self, object_path)
         self._object_path = object_path
+
+
+class AdvancedInterfaceTemplate(InterfaceTemplate, PropertiesInterface):
+    """Advanced template for DBus interface.
+
+    The interface provides the support for the standard interface
+    org.freedesktop.DBus.Properties.
+
+    Usage:
+
+        def connect_signals(self):
+            super().connect_signals()
+
+            self.implementation.properties_changed.connect(self.flush_changes)
+            self.implementation.x_changed.connect(self.changed("X"))
+
+        @property
+        def X(self, x) -> Int:
+            return self.implementation.x
+
+        @emits_properties_changed
+        def SetX(self, x: Int):
+            self.implementation.set_x(x)
+
+    """
+
+    def __init__(self, implementation):
+        PropertiesInterface.__init__(self)
+        InterfaceTemplate.__init__(self, implementation)
+
+    def changed(self, property_name):
+        """Returns a callback for the changed property.
+
+        :param property_name: a name of a DBus property
+        :return: a callback that should be run when the property has changed
+        """
+        self._properties_changes.check_property(property_name)
+        return lambda *args, **kwargs: self.report_changed_property(property_name)
```

Comment:
The signals of the implementation can emit values if they want to. I don't want to restrict them, since we are only interested in whether they were emitted or not.

Reply:
Ok then. Thanks for explanation.